vertical-pod-autoscaler/pkg/recommender/routines/recommender.go

```diff
@@ -81,11 +83,16 @@ func (r *recommender) updateVPAs() {
 			})
 
 		}
+		had := vpa.HasRecommendation()
 		vpa.Recommendation = &vpa_types.RecommendedPodResources{containerResources}
 		// Set RecommendationProvided if recommendation not empty.
 		if len(vpa.Recommendation.ContainerRecommendations) > 0 {
 			vpa.Conditions.Set(vpa_types.RecommendationProvided, true, "", "")
+			if !had {
+				metrics_recommender.ObserveRecommendationLatency(vpa.Created)
```

Comment:
I am not super sure about this metric. If there is no recommendation ever provided for a vpa we will not know from this metric, right? What we really need is information - how long has this vpa gone without a recommendation (and does it already have it or is it still waiting). 
Since admission controller only sees the vpa again once it is modified, maybe this should be moved into the recommender, as it looks at all vpas every loop? 
Open to discussion, I'd like this metric to give some meaningful signal on how well the system is performing.

Reply:
This metric is there to only measure time it took to provide the first recommendation.
As for tracing the number of VPAs w/o recommendation one could use vpa_objects_count.